sechub-integrationtest/src/test/resources/pds/checkmarx-wrapper/checkmarx-pds-job1.json

```diff
@@ -0,0 +1,36 @@
+{
+    "apiVersion" : "1.0",
+    "sechubJobUUID": "288607bf-ac81-4088-842c-005d5702a9e9", 
+    "productId": "PDS_CHECKMARX_INTEGRATIONTEST",
+    "parameters": [
+        {
+            "key": "pds.scan.configuration",
+            "value": "{\n  \"projectId\": \"sechub_evaluation\",\n  \"codeScan\": {\n    \"fileSystem\": {\n      \"folders\": [\n        \"weaknesses/\"\n      ]\n    }\n  }\n}\n"
```

Comment:
I am not sure, whether there is already a job using the new data section format. In case there is no test, having an integration test with the new data section format would be beneficial.

Reply:
## Current testing situation
There are already integration tests using the new data section format - For example: 

**Example1:**
`PDSCheckmarxIntegrationScenario17IntTest` does it by using the `IntegrationTestTemplateFile` enumeration.
The used Checkmarx mock adapter will inspect the uploaded source code folder path and return a dedicated mock.
The uploaded content is not inspected and does not contain a result. So only path resolving is tested here.

**Example2:**
`PDSBinaryAndSourceCodeHandlingScenario14IntTest` 
does upload source code zip file, extract it and send the results back as findings dynamically.

Example 2 does check the general auto extraction behavior for PDS and new data section
Example 1 does check the file structure given to PDS (checkmarx).

The PDS extraction behaviour is always same - and tested multiple times. In combination with the fact, that

the wrapper relies on the PDS default environment variables wrong usage is nearly impossible (no parameters, so only `java -jar -Dspring.profiles.active=defaul $TOOL_FOLDER/sechub-pds-wrapper-checkmarx.jar  ` is necessary).

## Possibility
If would want to test checkmarx cli wrapper in dynamic way, we would need to change the mock adapter behaviour to provide multiple results - e.g. by a special marker keyword in a reference-id. Than we could change override the checkmarx mock behavior  in this case and use a checkmarx-result.xml file inside the upload sources (for example) and return this.

But... This would 
- change the current mock implementation (I tried to change nothing inside the checkmarx adapter implementation at all) and only use the current implementation in PDS
- makes some effort
- could be done later if really necessary

## Conclusion
I think this is too much effort and the existing tests are enough.